sklearn/utils/deprecation.py

```diff
@@ -76,6 +79,22 @@ def wrapped(*args, **kwargs):
 
         return wrapped
 
+    def _decorate_instance(self, instance):
+        cls = instance.__class__
+        msg = "Thing %s is deprecated" % cls.__name__
+        if self.extra:
+            msg += "; %s" % self.extra
+
+        def wrapped(*args, **kwargs):
+            warnings.warn(msg, category=DeprecationWarning)
+            return instance(*args, **kwargs)
+
+        wrapped.__name__ = cls.__name__
+        wrapped.__dict__ = instance.__dict__
```

Comment:
Surely this isn't enough to make it behave like an object of that class in cases other than `__call__`. I don't see how `_decorate_instance` is applicable in the general case, and hence why we're writing something general rather than simply using a mixin or something.